artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,347 @@
+	}
+	transferAction = DownloadFiles
+
+	return
+}
+
+// Download the plugin files from the given url to the target directory (create path if needed or override existing files)
+func DownloadFiles(src string, pluginDir string, bundle PluginFiles) (err error) {
+	for _, file := range bundle {
+		fileName, fileDirs := file.SplitNameAndDirs()
+		srcURL := file.toURL(src)
+		dstDirPath := fileDirs.toPath(pluginDir)
+		log.Debug(fmt.Sprintf("transferring '%s' from '%s' to '%s'", fileName, fileDirs.toURL(src), dstDirPath))
+		if err = fileutils.CreateDirIfNotExist(dstDirPath); err != nil {
+			return
+		}
+		if err = downloadutils.DownloadFile(filepath.Join(dstDirPath, fileName), srcURL); err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Copy the plugin files from the given source to the target directory (create path if needed or override existing files)
+func CopyFiles(src string, pluginDir string, bundle PluginFiles) (err error) {
+	for _, file := range bundle {
+		fileName, fileDirs := file.SplitNameAndDirs()
+		srcPath := filepath.Join(src, fileName)
+		dstDirPath := fileDirs.toPath(pluginDir)
+		log.Debug(fmt.Sprintf("transferring '%s' from '%s' to '%s'", fileName, src, dstDirPath))
+		if err = fileutils.CreateDirIfNotExist(dstDirPath); err != nil {
+			return
+		}
+		if err = fileutils.CopyFile(dstDirPath, srcPath); err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Send reload command to the artifactory in order to reload the plugin files
+func sendReLoadCommand(servicesManager artifactory.ArtifactoryServicesManager) error {
+	log.Debug("reloading plugins")
+	serviceDetails := servicesManager.GetConfig().GetServiceDetails()
+	httpDetails := serviceDetails.CreateHttpClientDetails()
+
+	resp, body, err := servicesManager.Client().SendPost(serviceDetails.GetUrl()+pluginReloadRestApi, []byte{}, &httpDetails)
+	if err != nil {
+		return err
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (tic *InstallPluginCommand) Run() (err error) {
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold(fmt.Sprintf("Installing '%s' plugin...", tic.pluginName))))
+
+	// Initialize and validate
+	serviceManager, err := utils.CreateServiceManager(tic.targetServer, -1, 0, false)
+	if err != nil {
+		return
+	}
+	if _, err = validateMinArtifactoryVersion(serviceManager); err != nil {
+		return
+	}
+	// Get source, destination and transfer action
+	dst, err := tic.getPluginDirDestination()
+	if err != nil {
+		return
+	}
+	src, transferAction, err := tic.getTransferSourceAndAction()
+	if err != nil {
+		return
+	}
+	// Execute transferring action
+	if err = transferAction(src, dst.toPath(), tic.transferManger.files); err != nil {
+		return
+	}
+	// Reload plugins
+	if err = sendReLoadCommand(serviceManager); err != nil {
+		return
+	}
+
+	log.Info(coreutils.PrintTitle(coreutils.PrintBold("Plugin was installed successfully!.")))
```

Comment:
Let's rephrase to something like this:
`Plugin {tic.pluginName} installed successfully in the local Artifactory server.`

Reply:
done